src/service/resources-impl.js

```diff
@@ -407,13 +410,29 @@ export class Resources {
    * @private
    */
   buildReadyResources_() {
+    // Avoid cases where elements add more elements inside of them
+    // and cause an infinite loop of building - see #3354 for details.
+    if (this.isCurrentlyBuildingPendingResources_) {
```

Comment:
I've also tested that the new added resource is being built as suggested by @jridgewell below.